src/main/java/toughasnails/temperature/BlockTemperatureData.java

```diff
@@ -0,0 +1,14 @@
+package toughasnails.temperature;
+
+public class BlockTemperatureData {
+	public String blockName;
+	public float blockBaseTemperature;
+	public BlockTemperatureStateData[] stateTemperatures;
```

Comment:
I'm in the process of changing to use IBlockStates. It now occurs to me that using an explicit blockstate would require the user to specify a lot of redundant data.

e.g. The toughasnails:campfire, which has properties "age" and "burning" wants to have temperature 12 when burning and 0 when not burning regardless of age.

To achieve this when specifying IBlockStates it would be necessary to specify temperatures as

toughasnails:campfire[age=0,burning=true] = 12
toughasnails:campfire[age=1,burning=true] = 12
toughasnails:campfire[age=2,burning=true] = 12
...
toughasnails:campfire[age=7,burning=true] = 12

Am I missing something?

The first system I implemented allows specification of a temperature on a single property. This is admittedly less powerful, but I believe that the need to provide a full enumeration of all equivalent states is more painful and error prone than adding a way to specify AND on a set of properties in the first system.

I do agree that whatever is implemented the data should be validated to check that it actually specifies extant blocks and states. This will require moving this from mod preInit to mod init so that all mods have actually had a chance to create their blocks and states.